### DISCUSSION TITLE
Questions about the effectiveness of the models.

### DISCUSSION THREAD

**Discussion by DonTheApex** [brianpetro/obsidian-smart-connections#248] [Q&A]
How do different ChatGPT models, such as 3.5 turbo 4k, 16k, and ChatGPT 4, impact the performance of Smart Connections? Are there noticeable improvements in response quality that justifies the price increase? Does vault size impact effectiveness, and is it worth upgrading from 3.5 turbo 4k to 16k or ChatGPT 4? Your insights are greatly appreciated. Thank you!

**Comment by brianpetro** [comment]
The models you mentioned only apply to the Smart Chat feature. 

`gpt-4` (8k token limit) isn't yet widely available, but has produced much better results that `gpt-3.5-turbo` (4K token limit). 

However, the new `gpt-3.5-turbo-16k` (16K token limit) seems to be a great option for the Smart Connections Smart Chat feature. This is because it allows the most context sourced from your notes. So while `gpt-4` does the best job "reading between the lines" the `gpt-3.5-turbo-16k` model has room to include additional lines that make "reading between them" unnecessary.

There's also a price trade-offs between them, but I suggest testing the models out for yourself and monitoring your OpenAI usage costs to determine which option makes the most sense for you. 

I'd be interested to hear what you think after trying the different models. And maybe someone else could learn from your experience if you share it here.

Thanks for trying out Smart Connections
Brian 🌴
